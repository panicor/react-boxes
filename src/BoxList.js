import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

function BoxList() {
  let [boxes, setBoxes] = useState([]);

  let addBox = (boxObj) => {
    setBoxes((boxes) => [...boxes, boxObj]);
  };
  let removeBox = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  let boxComps = boxes.map((box) => (
    <Box key={box.id} id={box.id} width={box.width} height={box.height} backgroundColor={box.backgroundColor} handleRemove={removeBox}/>
  ));


  return(
    <div>
        <NewBoxForm addBox={addBox}/>
        {boxComps}
    </div>
  )
}


export default BoxList;
