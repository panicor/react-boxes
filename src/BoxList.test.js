import React from "react";
import { render, fireEvent, screen } from '@testing-library/react';
import BoxList from './BoxList';


function newBox(height=300, width=300, backgroundColor="green"){
    let heightInput = screen.getByLabelText("Height");
    let widthInput = screen.getByLabelText("Width");
    let backgroundColorInput = screen.getByLabelText("Background Color");

    fireEvent.change(heightInput, {target: {value: height}})
    fireEvent.change(widthInput, {target: {value: width}})
    fireEvent.change(backgroundColorInput, {target: {value: backgroundColor}})

    let btn = screen.getByText("Add box")

    fireEvent.click(btn);

}

test('renders', () => {
  render(<BoxList />);
});

test("snapshot matches", () => {
  let {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
})

test("adds new box", () => {
    let view = render(<BoxList />);
    expect(screen.queryByText("Remove")).not.toBeInTheDocument()
    newBox(view);

    let removeBtn = screen.getByText("Remove");
    expect(removeBtn).toBeInTheDocument();

})

test("removes box", () => {
    let view = render(<BoxList />);
    newBox(view);
  
    let removeBtn = screen.getByText("Remove");

    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
})