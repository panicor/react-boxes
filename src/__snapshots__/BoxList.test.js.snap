// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="height"
          >
            Height
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="width"
          >
            Width
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="backgroundColor"
          >
            Background Color
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="text"
            value=""
          />
        </div>
        <button
          id="boxBtn"
        >
          Add box
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
